parlai/tasks/dailydialog/agents.py

```diff
@@ -92,6 +92,13 @@ def share(self):
 
 class NoSilenceTeacher(Convai2Teacher):
     # Same as the default teacher, but it doesn't contain __SILENCE__ entries.
+    def __init__(self, opt, shared=None):
+        super().__init__(opt, shared)
+
+        # Calculate the correct number of examples.
+        dialogs = [d['dialogue'] for d in self.data]
+        self.num_exs -= len([0 for d in dialogs if len(d) >= 2])
```

Comment:
Can we use an explicit sum here?

I think it's `self.num_exs = sum(len(d['dialogue']) - 1 for d in self.data))`?

Reply:
Done!